src/main/java/com/habicus/core/model/Goals.java

```diff
@@ -0,0 +1,184 @@
+/*
+ _   _       _     _
+| | | | __ _| |__ (_) ___ _   _ ___
+| |_| |/ _` | '_ \| |/ __| | | / __|
+|  _  | (_| | |_) | | (__| |_| \__ \
+|_| |_|\__,_|_.__/|_|\___|\__,_|___/
+
+ * This file is part of the Habicus Core Platform (https://github.com/Habicus/Habicus-Core).
+ * Copyright (c) 2018 Habicus Core
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, version 3.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.habicus.core.model;
+
+import java.sql.Timestamp;
+import java.util.Objects;
+import javax.persistence.Basic;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.IdClass;
+
+@Entity
+@IdClass(GoalsPK.class)
```

Comment:
Does there need to be a composite primary key for goals? Would the goal ID not be enough for being unique?

Reply:
I'm willing to revisit this later once we refine our structure for the DB. Some recommendations were given for making a composite table to speed up searching. This was also auto-generated. I say we leave for now and might see benefits here to avoid table joins in the future.